dss-apps/dss-datamodel-center/dss-datamodel-measure-management/src/main/java/com/webank/wedatasphere/dss/datamodel/measure/service/impl/MeasureServiceImpl.java

```diff
@@ -0,0 +1,214 @@
+    }
+
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public int updateMeasure(Long id, MeasureUpdateVO vo) throws ErrorException {
+        DssDatamodelMeasure org = getBaseMapper().selectById(id);
+        if (org == null) {
+            LOGGER.error("errorCode : {}, update measure error not exists", ErrorCode.MEASURE_UPDATE_ERROR.getCode());
+            throw new DSSDatamodelCenterException(ErrorCode.MEASURE_UPDATE_ERROR.getCode(), "update measure error not exists");
+        }
+
+        //校验引用情况
+        if(datamodelReferencService.measureReferenceCount(vo.getName())>0
+                ||datamodelReferencService.measureReferenceCount(vo.getFieldIdentifier())>0
+                ||datamodelReferencService.measureReferenceCount(org.getName())>0
+                ||datamodelReferencService.measureReferenceCount(org.getFieldIdentifier())>0){
+            LOGGER.error("errorCode : {}, measure name can not be referenced ", ErrorCode.MEASURE_UPDATE_ERROR.getCode());
+            throw new DSSDatamodelCenterException(ErrorCode.MEASURE_UPDATE_ERROR.getCode(), "measure name can not be referenced");
+        }
+
+        //当更新名称时
+        if (!StringUtils.equals(vo.getName(), org.getName())) {
+            int repeat = getBaseMapper().selectCount(Wrappers.<DssDatamodelMeasure>lambdaQuery().eq(DssDatamodelMeasure::getName, vo.getName()));
+            if (repeat > 0 ) {
+                LOGGER.error("errorCode : {}, measure name can not repeat ", ErrorCode.MEASURE_UPDATE_ERROR.getCode());
+                throw new DSSDatamodelCenterException(ErrorCode.MEASURE_UPDATE_ERROR.getCode(), "measure name can not repeat ");
+            }
+        }
+
+        String orgFieldIdentifier = org.getFieldIdentifier();
+        //当更新标识时
+        if (!StringUtils.equals(vo.getFieldIdentifier(), orgFieldIdentifier)) {
+            int repeat = getBaseMapper().selectCount(Wrappers.<DssDatamodelMeasure>lambdaQuery().eq(DssDatamodelMeasure::getFieldIdentifier, vo.getFieldIdentifier()));
+
+            if (repeat > 0 || datamodelReferencService.measureReferenceCount(orgFieldIdentifier)>0) {
+                LOGGER.error("errorCode : {}, measure field identifier can not repeat or referenced", ErrorCode.MEASURE_UPDATE_ERROR.getCode());
+                throw new DSSDatamodelCenterException(ErrorCode.MEASURE_UPDATE_ERROR.getCode(), "measure field identifier can not repeat or referenced");
+            }
+        }
+        DssDatamodelMeasure updateOne = modelMapper.map(vo, DssDatamodelMeasure.class);
+        updateOne.setUpdateTime(new Date());
+        getBaseMapper().update(updateOne, Wrappers.<DssDatamodelMeasure>lambdaUpdate().eq(DssDatamodelMeasure::getId, id));
+        //同步atlas
+        UpdateModelTypeResult updateModelTypeResult = assertsSyncService.syncUpdateModel(
+                new UpdateModelEvent(this
+                        , DataModelSecurityContextHolder.getContext().getDataModelAuthentication().getUser()
```

Comment:
publisher.publishEvent( new UpdateModelEvent(this
                , DataModelSecurityContextHolder.getContext().getDataModelAuthentication().getUser()
                , vo.getFieldIdentifier()
                , orgFieldIdentifier
                , ClassificationConstant.MEASURE));